File: find_duplicates_array.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
  public:
    vector<int> duplicates(int arr[], int n) {
        // code here
        vector<int> c(n);
        for(int i=0; i<n; ++i) {
            c[arr[i]]++;
        }
        vector<int> ans;
        for(int i=0; i < n; ++i) {
            if (c[i] > 1) {
                ans.push_back(i);
            }
        }
        if (ans.size()==0) {
            return vector<int>{-1};
        }
        return ans;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t-- > 0) {
        int n;
        cin >> n;
        int a[n];
        for (int i = 0; i < n; i++) cin >> a[i];
        Solution obj;
        vector<int> ans = obj.duplicates(a, n);
        for (int i : ans) cout << i << ' ';
        cout << endl;
    }
    return 0;
}

// } Driver Code Ends

File: array/find_duplicates_array.py
class Solution:
    def duplicates(self, arr, n):
        # code here
        ans = set()
        OFFSET = 100000
        for i in range(n):
            b = arr[i] if arr[i] >= 0 else arr[i] + OFFSET
            if arr[b] < 0:
                ans.add(b)
            else:
                arr[b] -= OFFSET

        if len(ans) == 0:
            return [-1]

        return sorted(ans)


# {
# Driver Code Starts
if __name__ == "__main__":
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        res = Solution().duplicates(arr, n)
        for i in res:
            print(i, end=" ")
        print()


# } Driver Code Ends
